internal/scheduler/common.go

```diff
@@ -0,0 +1,278 @@
+		},
+	)
+	jobIdsByQueue := armadamaps.MapValues(
+		evictedJobsByQueue,
+		func(jobs []interfaces.LegacySchedulerJob) []string {
+			rv := make([]string, len(jobs))
+			for i, job := range jobs {
+				rv[i] = job.GetId()
+			}
+			return rv
+		},
+	)
+	return fmt.Sprintf(
+		"affected queues %v; resources %v; jobs %v",
+		maps.Keys(evictedJobsByQueue),
+		armadamaps.MapValues(
+			resourcesByQueue,
+			func(rl schedulerobjects.ResourceList) string {
+				return rl.CompactString()
+			},
+		),
+		jobIdsByQueue,
+	)
+}
+
+type AddOrSubtract int
+
+const (
+	Add AddOrSubtract = iota
+	Subtract
+)
+
+func UpdateUsage[S ~[]E, E interfaces.LegacySchedulerJob](
+	usage map[string]schedulerobjects.QuantityByPriorityAndResourceType,
+	jobs S,
+	priorityClasses map[string]configuration.PriorityClass,
+	addOrSubtract AddOrSubtract,
+) map[string]schedulerobjects.QuantityByPriorityAndResourceType {
+	if usage == nil {
+		usage = make(map[string]schedulerobjects.QuantityByPriorityAndResourceType)
+	}
+	for _, job := range jobs {
+		req := PodRequirementFromLegacySchedulerJob(job, priorityClasses)
+		if req == nil {
+			continue
+		}
+		requests := schedulerobjects.ResourceListFromV1ResourceList(req.ResourceRequirements.Requests)
+		queue := job.GetQueue()
+		m := usage[queue]
+		if m == nil {
+			m = make(schedulerobjects.QuantityByPriorityAndResourceType)
+		}
+		switch addOrSubtract {
+		case Add:
+			m.Add(schedulerobjects.QuantityByPriorityAndResourceType{req.Priority: requests})
+		case Subtract:
+			m.Sub(schedulerobjects.QuantityByPriorityAndResourceType{req.Priority: requests})
+		default:
+			panic(fmt.Sprintf("invalid operation %d", addOrSubtract))
+		}
+		usage[queue] = m
+	}
+	return usage
+}
+
+func jobSchedulingContextsFromJobs[T interfaces.LegacySchedulerJob](jobs []T, executorId string, priorityClasses map[string]configuration.PriorityClass) []*schedulercontext.JobSchedulingContext {
+	if jobs == nil {
+		return nil
+	}
+	if len(jobs) == 0 {
+		return make([]*schedulercontext.JobSchedulingContext, 0)
+	}
+	jctxs := make([]*schedulercontext.JobSchedulingContext, len(jobs))
+	timestamp := time.Now()
+	for i, job := range jobs {
+		jctxs[i] = &schedulercontext.JobSchedulingContext{
+			Created:    timestamp,
+			ExecutorId: executorId,
+			JobId:      job.GetId(),
+			Job:        job,
+			Req:        PodRequirementFromJobSchedulingInfo(job.GetRequirements(priorityClasses)),
+		}
+	}
+	return jctxs
+}
+
+func isEvictedJob(job interfaces.LegacySchedulerJob) bool {
+	return job.GetAnnotations()[schedulerconfig.IsEvictedAnnotation] == "true"
+}
+
+func targetNodeIdFromLegacySchedulerJob(job interfaces.LegacySchedulerJob) (string, bool) {
+	req := PodRequirementFromLegacySchedulerJob(job, nil)
+	nodeId, ok := req.NodeSelector[schedulerconfig.NodeIdLabel]
+	return nodeId, ok
+}
+
+func GangIdAndCardinalityFromLegacySchedulerJob(job interfaces.LegacySchedulerJob, priorityClasses map[string]configuration.PriorityClass) (string, int, bool, error) {
```

Comment:
I'd document this function to indicate what each of the return values mean.  Really though I'd do the following:

- Add `GetGangId` and `GetGangCardinality` (and possibly `isGangJob`) functions to the LegacySchedulerJob interface
- For `SchedulerJob` model these as first class fields with no annotation lookups necessary
- For api.job use the logic here.